ChessBoard.UnitTests/ChessBoardTests.cs

```diff
@@ -0,0 +1,82 @@
+using System.Text;
+using Microsoft.Extensions.Logging;
+using Moq;
+
+namespace ChessBoardApp.ChessBoard.Tests
+{
+    class TestLogger<T> : ILogger<T>
+    {
+        public List<string> LoggedMessages { get; } = new();
+
+        public IDisposable BeginScope<TState>(TState state) => null;
+
+        public bool IsEnabled(LogLevel logLevel) => true;
+
+        public void Log<TState>(
+            LogLevel logLevel,
+            EventId eventId,
+            TState state,
+            Exception exception,
+            Func<TState, Exception, string> formatter
+        )
+        {
+            string logMessage = formatter(state, exception);
+            LoggedMessages.Add(logMessage);
+        }
+    }
+
+    public class ChessBoardGeneratorTests
+    {
+        [Fact]
+        public void GenerateRectangleBoard_ShouldLogCorrectChessBoard()
```

Comment:
Good attempt, but let's consider https://adform.slack.com/archives/C05ECTFN5GE/p1688356759903009?thread_ts=1688335639.652929&cid=C05ECTFN5GE points 2 and 3